backend/btrixcloud/migrations/migration_0045_crawl_counts.py

```diff
@@ -0,0 +1,56 @@
+"""
+Migration 0045 - Recalculate crawl filePageCount and errorPageCount
+"""
+
+from btrixcloud.migrations import BaseMigration
+
+
+MIGRATION_VERSION = "0045"
+
+
+# pylint: disable=duplicate-code
+class Migration(BaseMigration):
+    """Migration class."""
+
+    # pylint: disable=unused-argument
+    def __init__(self, mdb, **kwargs):
+        super().__init__(mdb, migration_version=MIGRATION_VERSION)
+
+        self.page_ops = kwargs.get("page_ops")
+
+    async def migrate_up(self):
+        """Perform migration up.
+
+        Recalculate crawl filePageCount and errorPageCount for all crawls
+        """
+        crawls_mdb = self.mdb["crawls"]
+
+        if self.page_ops is None:
+            print(
+                "Unable to reset crawl page counts, missing page_ops",
+                flush=True,
+            )
+            return
+
+        # Reset filePageCount and errorPageCount to 0 for all crawls
+        await crawls_mdb.update_many(
+            {},
+            {
+                "$set": {
+                    "filePageCount": 0,
+                    "errorPageCount": 0,
+                }
+            },
+        )
+
+        # Recalculate filePageCount and errorPageCount for every crawl
+        async for crawl_raw in crawls_mdb.find({}):
```

Comment:
I think we only need to run this for crawls where filePageCount > 0 or errorPageCount > 0, eg`{$or: [{"errorPageCount": {$gt: 0}}, {"filePageCount": {$gt: 0}}]}` and reset just for those crawls?

Reply:
Good point! This has been updated. I think to avoid needing to save a potentially very large list of crawl ids in memory, iterating through each matching crawl and handling them one at a time probably makes sense.